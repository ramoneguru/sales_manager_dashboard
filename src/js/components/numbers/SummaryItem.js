/**
 *
 * @author Drew Robinson (hello@drewrobinson.com)
 * @version 0.0.1
 * @param props
 * @exports SummaryItem Constructor
 */

import React from 'react';
import Constants from '../../constants/constants';
import PropTypes from 'prop-types';

/**
 * Represents a summary of an indivdual sales rep number of activities
 * @param props
 * @returns {XML}
 * @constructor
 * @desc prefer normal functions (not arrow functions) for stateless/refless components over over classes
 */
function SummaryItem( props ) {
  let rate = 0;
  let src = `${Constants.PROTOCOL}//${Constants.HOST}/images/${props.avatar}`

  if(props.data.length > 0){
    let perc = (props.data[0]['12M'][2]/(props.data[0]['12M'][0] + props.data[0]['12M'][1])).toFixed(2);
    rate = perc * 100;
  }
  return (
    <div className="rep-summary__item">
        <figure>
          <span><img src={src} alt={props.name} role="presentation"/></span>
          <figcaption>{props.name}</figcaption>
        </figure>
        <span>{rate}%</span>
    </div>
  )
}

SummaryItem.propTypes = {
  "name"   : PropTypes.string.isRequired,
  "avatar" : PropTypes.string.isRequired,
  "data"   : PropTypes.array.isRequired
};

export default SummaryItem;
