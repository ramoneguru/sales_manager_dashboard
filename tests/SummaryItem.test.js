import React from 'react';
import { shallow, render, mount } from 'enzyme';
import SummaryItem from '../src/js/components/numbers/SummaryItem';


describe('SummaryItem component ', function(){
  "use strict";

  it('should render correctly', () => {
    const name = 'mock'
    const avatar = 'mock'
    const data = []
    const item = mount(
      <SummaryItem name={ name } avatar={ avatar } data={ data } />
    );

    expect(item).toMatchSnapshot()
  });

});
