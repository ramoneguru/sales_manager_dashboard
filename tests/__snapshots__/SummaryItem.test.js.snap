// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryItem component  should render correctly 1`] = `
<SummaryItem
  avatar="mock"
  data={Array []}
  name="mock"
>
  <div
    className="rep-summary__item"
  >
    <figure>
      <span>
        <img
          alt="mock"
          role="presentation"
          src="about://undefined/images/mock"
        />
      </span>
      <figcaption>
        mock
      </figcaption>
    </figure>
    <span>
      0
      %
    </span>
  </div>
</SummaryItem>
`;
